setup.py

```diff
@@ -323,15 +323,29 @@ def get_long_description():
             "PySide2;python_version<'3.9'",
             "traitsui;python_version<'3.9'",
         ],
+        "examples": [
+            # dependencies for examples
+            "numpy",
+            "pillow",
+        ]
     },
     ext_modules=[setuptools.Extension("traits.ctraits", ["traits/ctraits.c"])],
     package_data={
+        "traits": [
+            "examples/introduction/*",
+            "examples/introduction/*/*",
+        ],
         "traits.tests": [
             "test-data/historical-pickles/README",
             "test-data/historical-pickles/*.pkl",
             "test-data/historical-pickles/*.py",
         ],
     },
+    entry_points={
+        "etsdemo_data": [
+            "introduction = traits._entry_point:introduction",
```

Comment:
Traits can name these entry points however it wants. It can have many entry points if it likes.
I don't think a common format brings much values, in fact, it might be tempting to make assumptions about these names being consistently used.